examples/double_gyre.jl

```diff
@@ -0,0 +1,285 @@
+using Oceananigans.Grids: xnode, ynode, znode
+using Oceananigans.TurbulenceClosures: VerticallyImplicitTimeDiscretization
+using CairoMakie
+using Printf
+
+const Lx = 4000kilometers # east-west extent [m]
+const Ly = 6000kilometers # north-south extent [m]
+const Lz = 1.8kilometers  # depth [m]
+
+Δt₀ = 10minutes
+stop_time = 10years
+
+Nx = 160
+Ny = 240
+Nz = 50
+
+σ = 1.2 # stretching factor
+hyperbolically_spaced_faces(k) = - Lz * (1 - tanh(σ * (k - 1) / Nz) / tanh(σ))
+
+grid = RectilinearGrid(CPU();
+                       size = (Nx, Ny, Nz),
+                          x = (-Lx/2, Lx/2),
+                          y = (-Ly/2, Ly/2),
+                          z = hyperbolically_spaced_faces,
+                   topology = (Bounded, Bounded, Bounded))
+
+# We plot vertical spacing versus depth to inspect the prescribed grid stretching:
+
+fig = Figure(resolution=(1200, 800))
+ax = Axis(fig[1, 1], ylabel = "Depth (m)", xlabel = "Vertical spacing (m)")
+lines!(ax, grid.Δzᵃᵃᶜ[1:grid.Nz], grid.zᵃᵃᶜ[1:grid.Nz])
+scatter!(ax, grid.Δzᵃᵃᶜ[1:Nz], grid.zᵃᵃᶜ[1:Nz])
+
+save("double_gyre_grid_spacing.svg", fig)
+nothing #hide
+
+# ![](double_gyre_grid_spacing.svg)
+
+α  = 2e-4 # [K⁻¹] thermal expansion coefficient
+g  = 9.81 # [m s⁻²] gravitational constant
+cᵖ = 3991 # [J K⁻¹ kg⁻¹] heat capacity for seawater
+ρ₀ = 1028 # [kg m⁻³] reference density
+
+parameters = (Ly = Ly,
+              Lz = Lz,
+               τ = 0.1 / ρ₀,     # surface kinematic wind stress [m² s⁻²]
+               μ = 1 / 30days,   # bottom drag damping time-scale [s⁻¹]
+              Δb = 30 * α * g,   # surface vertical buoyancy gradient [s⁻²]
+               λ = 30days        # relaxation time scale [s]
+              )
+
+# ## Boundary conditions
+#
+# ### Wind stress
+
+@inline u_stress(x, y, t, p) = - p.τ * cos(2π * y / p.Ly)
+u_stress_bc = FluxBoundaryCondition(u_stress; parameters)
+
+# ### Bottom drag
+@inline u_drag(x, y, t, u, p) = - p.μ * p.Lz * u
+@inline v_drag(x, y, t, v, p) = - p.μ * p.Lz * v
+
+u_drag_bc = FluxBoundaryCondition(u_drag; field_dependencies=:u, parameters)
+v_drag_bc = FluxBoundaryCondition(v_drag; field_dependencies=:v, parameters)
+
+u_bcs = FieldBoundaryConditions(top = u_stress_bc, bottom = u_drag_bc)
+v_bcs = FieldBoundaryConditions(                   bottom = v_drag_bc)
+
+
+# ### Buoyancy relaxation
+
+@inline buoyancy_relaxation(x, y, z, t, b, p) = - 1 / p.λ * (b - p.Δb * y / p.Ly)
+Fb = Forcing(buoyancy_relaxation; parameters, field_dependencies=:b)
+
+# ## Turbulence closure
+horizontal_diffusive_closure = HorizontalScalarDiffusivity(ν=5000, κ=1000)
+vertical_diffusive_closure = VerticalScalarDiffusivity(VerticallyImplicitTimeDiscretization(),
+                                                       ν=1e-2, κ=1e-5)
+
+# ## Model building
+
+model = HydrostaticFreeSurfaceModel(; grid,
+                                    free_surface = ImplicitFreeSurface(),
+                                    momentum_advection = WENO(),
```

Comment:
```suggestion
        momentum_advection = VectorInvariant(vorticity_scheme  = WENO(grid),
                                             divergence_scheme = WENO(grid),
                                             vertical_scheme   = WENO(grid))
```

might need some reformatting